src/util.c

```diff
@@ -648,15 +648,17 @@ netplan_get_id_from_nm_filepath(const char* filename, const char* ssid, char* ou
 
     /* Move pointer to start of netplan ID inside filename string */
     start = pos + strlen(nm_prefix);
-    id_len = end - start;
+    g_assert(end - start > 0);
+    id_len = (gsize)(end - start);
```

Comment:
nitpick: If we do an explicit typecast anyway. Should we declare `id_len` as `ssize_t` directly above?

Reply:
Technically `end - start` is a `ptrdiff_t` (`signed long`). Changing the type of `id_len` causes conversion issues in other lines. As I added the assertion before, I think it's ok to leave the cast.

Reply:
correction, `ptrdiff_t` is a `long` on 64bit archs, on 32bit (tested on armhf) it's an int.